lib/ansible/plugins/callback/log_plays.py

```diff
@@ -51,8 +60,13 @@ def __init__(self):
 
         super(CallbackModule, self).__init__()
 
-        if not os.path.exists("/var/log/ansible/hosts"):
-            os.makedirs("/var/log/ansible/hosts")
+    def set_options(self, task_keys=None, var_options=None, direct=None):
+        super(CallbackModule, self).set_options(task_keys=task_keys, var_options=var_options, direct=direct)
+
+        self.log_folder = self.get_option("log_folder")
+
+        if not os.path.exists(self.log_folder):
+            os.makedirs(self.log_folder)
```

Comment:
this can have race conditions use makedirs_safe to minimize (not solve) the issue

Reply:
Thanks again; and done.